Sources/GRPC/GRPCClientStateMachine.swift

```diff
@@ -346,6 +346,21 @@ struct GRPCClientStateMachine {
     }
   }
 
+  /// Receive a DATA frame with the end stream flag set. Determines whether it is safe for the
+  /// caller to ignore the end stream flag or whether a synthesised status should be forwarded.
+  ///
+  /// Receiving a DATA frame with the end stream flag set is unexpected: the specification dictates
+  /// that an RPC should be ended by the server sending the client a HEADERS frame with end stream
+  /// set. However, we will tolerate end stream on a DATA frame providing we have received trailers
+  /// containing 'grpc-status' (i.e. we are in the 'clientClosedServerClosed' state). In cases where
+  /// we don't expect end of stream on a DATA frame we will emit a status with a message explaining
+  /// the protocol violation.
```

Comment:
I'm a bit confused about this state description. When will we receive a DATA frame _after_ having received a HEADERS frame that contains trailers? NIOHTTP2 enforces that the trailers header block must have END_STREAM, so no DATA frame on that stream may validly follow it. What have I missed?

Reply:
Ah, right, that's a good point. That was me assuming a state in which it could be valid. Really the case we tolerate it is "the rpc is already complete".